src/main/scala/com/ubirch/services/key/PubKeyVerificationService.scala

```diff
@@ -76,14 +76,19 @@ class DefaultPubKeyVerificationService @Inject() (jsonConverter: JsonConverterSe
   }
 
   def validate(publicKey: Array[Byte], signature: Array[Byte], message: Array[Byte], curve: Curve): Boolean = {
+
+    def validateBase(signature: Array[Byte]): Boolean = GeneratorKeyFactory
+      .getPubKey(publicKey, curve)
+      .verify(message, signature)
+
     try {
-      GeneratorKeyFactory
-        .getPubKey(publicKey, curve)
-        .verify(message, signature)
+      validateBase(signature)
     } catch {
       case e: InvalidKeySpecException =>
         logger.error("Failed to decode 2 -> exception={} message={}", e.getClass.getCanonicalName, e.getMessage)
         false
+      case _: SignatureException =>
+        validateBase(Utils.pointsToASN1(signature))
```

Comment:
What happens if the second `validateBase` call fails with `SignatureException` again ?

Reply:
I see 👍